### ISSUE TITLE
Update to final CQL2 spec

### ISSUE THREAD

**Issue by philvarner** [stac-api-extensions/filter#18]
This is waiting on the OGC release of CQL2. 

Status as of Sept 2023:

- https://github.com/opengeospatial/ogcapi-features/issues/824#issuecomment-1738207283

**Comment by emmanuelmathot** [comment]
Are we waiting this release to update the [`InBetweenPredicate`](https://github.com/opengeospatial/ogcapi-features/blob/master/cql2/standard/schema/cql2.yml#L132C5-L132C23) [example](https://github.com/stac-api-extensions/filter#example-11-using-between) ?
It should be a 3 items array operator:
```json
{
  "filter-lang": "cql2-json",
  "filter": {
    "op": "between",
    "args": [
      { "property": "eo:cloud_cover" },
      0, 50
    ]
  }
}
```

**Comment by philvarner** [comment]
No, that was just overlooked.

**Comment by Konkrad** [comment]
Any update on this? The Spec seems to be public https://www.ogc.org/publications/standard/cql2/

**Comment by philvarner** [comment]
@Konkrad no one is currently funding development of the spec, so this is on hold until someone decides to do that.
